api/auth.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+message AuthAction {
+
+  enum VerifyType {
+    option allow_alias = true;
+
+    DEFAULT = 0;
+    // Request matches all rules are allowed, otherwise denied.
+    ALLOW = 0;
+    // Request matches all rules or missing required auth fields are denied,
+    // otherwise allowed.
+    DENY = 1;
+  }
+
+  VerifyType verify_type = 1;
+
+  // Check peer identity using X.509 certificate.
+  message X509 {
+    // SHA256 hash of peer certificate.
+    string sha256_hash = 1;
+    // Subject of peer certificate.
+    string subject = 2;
+    // Subject alt name of peer certificate.
+    string subject_alt_name = 3;
+  }
+
+  message X509VerifyRule {
+    // List of X509 to check against.
+    repeated X509 x509s = 2;
+  }
+
+  // Choose among multiple verify rules.
+  message VerifyRule {
+    oneof verify_rule {
+      X509VerifyRule x509_verify_rule = 1;
+    }
+  }
+
+  // List of verify rules, all should match.
+  repeated VerifyRule verify_rules = 2;
```

Comment:
Is there a case for a combined white/blacklist, or do we just want one of these types in practice?

Reply:
have both white/black list at the same time maybe confusing.

i think we can have a *default_action* with allow/deny, and each rule have its own *action*.